gemma-core/src/main/java/ubic/gemma/core/util/StopWatchMonitor.java

```diff
@@ -0,0 +1,33 @@
+package ubic.gemma.core.util;
+
+import org.apache.commons.lang3.time.StopWatch;
+
+/**
+ * Start or resume a {@link StopWatch} with a try-with-resource statement.
+ *
+ * This is really handy to measure code regions and report time usage.
+ *
+ * @author poirigui
+ */
+public class StopWatchMonitor implements AutoCloseable {
```

Comment:
I'm thinking of using a different name for this tool. It basically records time for a code region using a try-with-resource statement. It's super-convenient when you want to measure the cumulative time consumed by a statement in a loop.